src/content/docs/dropins/all/extending.mdx

```diff
@@ -12,50 +12,347 @@ import { Steps } from '@astrojs/starlight/components';
+```js
+const GiftOptionsField = document.createElement('gift-options-field');
+GiftOptionsField.setAttribute('loading', 'true');
+```
 
-### Container
+Then, add the container to the layouts of the checkout drop-in. The following example shows how to add the container to the `checkout` layout.
 
-Component that manages or encapsulates other components. Containers handle logic, fetch data, manage state, and pass data to the UI components that are rendered on the screen.
+```js title='commerce-checkout.js'
+function renderMobileLayout(block) {
+  root.replaceChildren(
+    heading,
+    giftOptionsField,
+    ...
+  );
+
+  block.replaceChildren(root);
+}
+
+function renderDesktopLayout(block) {
+  main.replaceChildren(
+    heading,
+    giftOptionsField,
+    ...
+  );
+
+  block.replaceChildren(block);
+}
+```
 
-### Slot
+</Task>
 
-Component that provides placeholders to add other components. You can use a drop-in component's built-in slots to add or remove UI components and functions. Or you can add your own additional Slots.
+<Task>
+### Add handler for gift message submission
 
-### Component
+Now that we have the UI in place, we need to add a handler to submit the gift message to the Commerce API. We'll use the `fetch` API to send a GraphQL mutation to update the gift message.
 
-Overloaded term in web development. Everything is a component. It's components all the way down. This is why we need to be specific about what kind of component we are talking about. For example, from top-to-bottom, big-to-small, a **drop-in component** can contain multiple **container components** that can contain multiple **slot components** that can contain multiple **UI components**.
+```js title='commerce-checkout.js'
+giftOptionsField.submitGiftMessageHandler = async (event) => {
+  event.preventDefault();
 
-</Vocabulary>
+  const form = event.target;
+  const formData = new FormData(form);
+  const cartId = formData.get('cartId');
+  const fromName = formData.get('fromName');
+  const toName = formData.get('toName');
+  const giftMessage = formData.get('giftMessage');
 
-## Extend with third-party components
+  giftOptionsField.setAttribute('loading', 'true');
+  console.log('form data', cartId, fromName, toName, giftMessage);
 
-This tutorial describes the steps needed to extend a drop-in by integrating a fictitious third-party ratings & reviews component in the product details drop-in.
+  const giftMessageInput = {
+    from: fromName,
+    to: toName,
+    message: giftMessage,
+  }
 
-:::note[Performance]
-Be vigilant about how you display and fetch data for your third-party component to ensure you avoid any Cumulative Layout Shifts (CLS) that could significantly affect the site's core web vital scores.
-:::
+  fetchGraphQl(`
+    mutation SET_GIFT_OPTIONS($cartId: String!, $giftMessage: GiftMessageInput!) {
+        setGiftOptionsOnCart(input: {
+          cart_id: $cartId,
+          gift_message: $giftMessage
+          gift_receipt_included: false
```

Comment:
I wouldn't include `gift_receipt_included` or `printed_card_included` fields, as they're outside the scope of the example. If you mention them, then I feel like you need to expand the scope to include those fields in the UI.